packages/mobile/src/tokens/saga.ts

```diff
@@ -248,3 +253,23 @@ export async function getCurrencyAddress(currency: Currency) {
       return contractKit.registry.addressFor(CeloContract.StableTokenEUR)
   }
 }
+
+export function* importTokenInfo() {
+  const tokens: Token[] = yield readOnceFromFirebase('tokensInfo')
+  let token: Token
+  const kit: ContractKit = yield call(getContractKit)
+  const address: string = yield select(currentAccountSelector)
+  const balances: TokenBalances = {}
+  for (token of tokens) {
```

Comment:
Lets do this in parallel so its faster :D

To do that I think we might need to extract the code inside the for loop to it's own function `fetchTokenBalance` and then:
```
const balances = yield all(tokens.map((token: Token) => call(fetchTokenBalance, token))
```

Lets also add a try/catch around each individual balance fetch so that if one fails we still get the data for all the others. Maybe just leave the balance as `null` in that case.